Source/WebKit/WebProcess/haiku/WebProcessMainHaiku.cpp

```diff
@@ -39,41 +40,37 @@ namespace WebKit {
 
 class WebProcessApp: public BApplication
 {
+	private:
+	int argc;
+	char** argv;
 	public:
-	WebProcessApp(void):BApplication("application/x-vnd.haiku-webkit.webprocess")
+	WebProcessApp(int argc,char** argv):BApplication("application/x-vnd.haiku-webkit.webprocess")
 	{
+		this->argc = argc;
+		this->argv = argv;
 	}
 
 	void MessageReceived(BMessage* msg)
 	{
-		fprintf(stderr, "%s\n", __PRETTY_FUNCTION__);
-		msg->PrintToStream();
+		//msg->PrintToStream();
+		switch(msg->what)
+		{
+			default:
+			BApplication::MessageReceived(msg);
+		}
 	}
 
 	void ReadyToRun()
 	{	
-		fprintf(stderr, "%s\n", __PRETTY_FUNCTION__);
-		RunLoop::initializeMainRunLoop();
-		RunLoop::run();
+		AuxiliaryProcessMain<WebProcess>(argc,argv);
```

Comment:
This seems a bit backwards. I think AuxiliaryProcessMain should be in charge of creating a BApplication, if that's possible.

Reply:
Yes AuxiliaryProcessMain creating will be much better as it will be easier to create a template BApllication so that we can use for both webprocess and network process